core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableInfo.java

```diff
@@ -291,6 +318,17 @@ public void write(DataOutput out) throws IOException {
       }
     }
     out.writeBoolean(isSchemaModified);
+
+    out.writeUTF(format);
+    boolean isFormatPropertiesExists = null != formatProperties && formatProperties.size() > 0;
+    out.writeBoolean(isFormatPropertiesExists);
+    if (isFormatPropertiesExists) {
+      out.writeShort(formatProperties.size());
```

Comment:
It is better to write `formateProperties` in one shot, performance is much better than writing each entry.

Reply:
The formatProperties is a javaMap that converted from scalaMap. When I try to directly write this map as binary, a serialization problem occurs.